pkg/util/validation.go

```diff
@@ -105,6 +105,27 @@ func IsValidPortNum(port int) bool {
 	return 0 < port && port < 65536
 }
 
+// IsValidPortString check references against
+func IsValidPortString(port string) bool {
+	// implemetns check of https://tools.ietf.org/html/rfc6335#section-5.1
+	if len(port) < 1 || len(port) > 15 {
+		// MUST be at least 1 character and no more than 15 characters long
+		return false
+	}
+	if regexp.MustCompile("--").MatchString(port) {
+		// hyphens MUST NOT be adjacent to other hyphens
+		return false
+	}
+
+	if regexp.MustCompile("^[A-Za-z0-9][A-Za-z0-9-]*[^-]$").MatchString(port) &&
```

Comment:
Would move the rexegs to private variables at the package level so you don't have to recompile every time.